### ISSUE TITLE
`use_linear_attn = True` produce noise and unstable loss

### ISSUE THREAD

**Issue by ken012git** [lucidrains/imagen-pytorch#55]
After moving from v0.0.60. to v0.1.10, I found the Imagen loss is unstable in the early training steps and the results is noisy from the early stage.

<img width="471" alt="Screen_Shot_2022-06-12_at_11 13 46_AM" src="https://user-images.githubusercontent.com/9960543/173449531-8f30f31f-8e65-4c1f-8862-14fcd8ec9bbe.png">

![unknown](https://user-images.githubusercontent.com/9960543/173449539-4e44e300-5cfa-47d6-b8d2-f9828ae691a6.png)

The problem is gone when I set `use_linear_attn = False`

**Comment by lucidrains** [comment]
@ken012git :pray: do you want to try `0.2.4`? i think i found the issue :man_facepalming:

**Comment by ken012git** [comment]
Sure! Thanks for your immediate response!

I would also like to know what causes the issue. =)

**Comment by lucidrains** [comment]
@ken012git forgot the residual 🤦 and also needed a feedforward after it anyways

**Comment by ken012git** [comment]
Hi @lucidrains ,

I have tested v0.2.4 and the issue seems gone. Thanks!

```
# test model, resolution 64
unet1 = Unet(
        dim = 32,
        cond_dim = 512,
        dim_mults = (1, 2, 4, 8),
        num_resnet_blocks = (2, 2, 2, 2),    # small
        layer_attns = (False, False, False, True),
        layer_cross_attns = (False, False, False, True),
       # use_linear_attn = False,        
        use_linear_attn = True,
    )
```
Loss curve, blue: `use_linear_attn =False`,red: `use_linear_attn =True`
<img width="468" alt="Screen Shot 2022-06-13 at 4 08 59 PM" src="https://user-images.githubusercontent.com/9960543/173462048-27fc3ea7-9260-44fd-8529-e3f6178a8c30.png">


early stage results, left: `use_linear_attn =False`,right: `use_linear_attn =True`
<img width="920" alt="Screen Shot 2022-06-13 at 4 10 25 PM" src="https://user-images.githubusercontent.com/9960543/173462159-06e13cf2-7038-465a-a317-3e43da6d24cf.png">


I am wondering we should use transformers or linear attention layers at this [line](https://github.com/lucidrains/imagen-pytorch/blob/main/imagen_pytorch/imagen_pytorch.py#L993) that configured by  `use_linear_attn`.

Would you point me relevant papers? Thanks

**Comment by lucidrains** [comment]
@ken012git thank you for the experiments! basically, in a lot of papers, researchers remove attention past a certain token length (1024 or 2048) since it is prohibitively expensive due to the quadratic problem. but i like to substitute them with linear attention, even if it is a bit weaker. my favorite linear attention remains https://arxiv.org/abs/1812.01243 , and here i am also giving it a depthwise conv recommended by the [primer paper](https://arxiv.org/abs/2109.08668)
